lib/RemoteControl/src/App.cpp

```diff
@@ -0,0 +1,310 @@
+ *****************************************************************************/
+
+/******************************************************************************
+ * Private Methods
+ *****************************************************************************/
+
+void App::fatalErrorHandler()
+{
+    /* Turn on Red LED to signal fatal error. */
+    Board::getInstance().getRedLed().enable(true);
+
+    while (true)
+    {
+        ;
+    }
+}
+
+void App::cmdTopicCallback(const String& payload)
+{
+    StaticJsonDocument<JSON_DOC_DEFAULT_SIZE> jsonPayload;
+    DeserializationError                      error = DeserializationError::Ok;
+
+    error = deserializeJson(jsonPayload, payload.c_str());
+
+    if (error != DeserializationError::Ok)
+    {
+        LOG_ERROR("JSON Deserialization Error %d.", error);
+    }
+    else
+    {
+        uint8_t payloadSize = sizeof(uint8_t);
+        uint8_t buffer[payloadSize];
+
+        buffer[0U] = jsonPayload["CMD_ID"].as<uint8_t>();
+
+        if (true == m_smpServer.sendData(CH_NAME_CMD, buffer, 1U))
+        {
+            LOG_DEBUG("Command %d sent.", buffer[0U]);
+        }
+        else
+        {
+            LOG_WARNING("Failed to send command %d.", buffer[0U]);
+        }
+    }
+}
+
+void App::motorSpeedsTopicCallback(const String& payload)
+{
+    StaticJsonDocument<JSON_DOC_DEFAULT_SIZE> jsonPayload;
+    DeserializationError                      error = DeserializationError::Ok;
+
+    error = deserializeJson(jsonPayload, payload.c_str());
+
+    if (error != DeserializationError::Ok)
+    {
+        LOG_ERROR("JSON Deserialization Error %d.", error);
+    }
+    else
+    {
+        uint8_t payloadSize = (2U * sizeof(int16_t));
+        uint8_t buffer[payloadSize];
+
+        Util::int16ToByteArray(&buffer[0U * sizeof(int16_t)], sizeof(int16_t), jsonPayload["LEFT"].as<int16_t>());
+        Util::int16ToByteArray(&buffer[1U * sizeof(int16_t)], sizeof(int16_t), jsonPayload["RIGHT"].as<int16_t>());
+
+        if (true == m_smpServer.sendData(CH_NAME_MOTOR_SPEEDS, buffer, payloadSize))
+        {
+            LOG_DEBUG("Motor speeds sent");
+        }
+        else
+        {
+            LOG_WARNING("Failed to send motor speeds");
+        }
+    }
+}
+
+/******************************************************************************
+ * External Functions
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Functions
+ *****************************************************************************/
+
+/**
+ * Receives remote control commands over SerialMuxProt channel.
+ *
+ * @param[in] payload       Command id
+ * @param[in] payloadSize   Size of command id
+ */
+void App_cmdRspChannelCallback(const uint8_t* payload, const uint8_t payloadSize)
+{
+    LOG_DEBUG("CMD_RSP: 0x%02X", payload[0]);
+}
+
+/**
+ * Receives line sensor data over SerialMuxProt channel.

```

Comment:
Wrong description.

Reply:
Really? It is the callback for SMP. Why is it wrong?